gst/nnstreamer/elements/gsttensor_trainer.c

```diff
@@ -570,39 +570,49 @@ gst_tensor_trainer_chain (GstPad * sinkpad, GstObject * parent,
   void *ptr;
 
   trainer = GST_TENSOR_TRAINER (parent);
+  in_flexible = gst_tensor_pad_caps_is_flexible (sinkpad);
+  num_tensors = gst_buffer_n_tensor (inbuf);
 
-  mem_blocks = gst_buffer_n_memory (inbuf);
-  for (i = 0; i < mem_blocks; i++) {
-    in_mem[i] = gst_buffer_peek_memory (inbuf, i);
+  for (i = 0; i < num_tensors; i++) {
+    in_mem[i] = gst_tensor_buffer_get_nth_memory (inbuf, i);
     if (!gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
       GST_ERROR_OBJECT (trainer, "Could not map in_mem[%u] GstMemory", i);
       goto error;
     }
-    in_flexible = gst_tensor_pad_caps_is_flexible (sinkpad);
+
     /* Get header size */
     header_size = 0;
     if (in_flexible) {
-      gst_tensor_meta_info_parse_header (&in_meta[i], &in_info[i].data);
+      if (!gst_tensor_meta_info_parse_header (&in_meta[i], in_info[i].data)) {
+        GST_ERROR_OBJECT (trainer, "Invalid Flexible tensors");
+        goto error;
+      }
+
       header_size = gst_tensor_meta_info_get_header_size (&in_meta[i]);
+      gst_tensor_meta_info_convert (&in_meta[i],
```

Comment:
For later work, need update for nth tensor-info ptr:
```
_info = gst_tensors_info_get_nth_info (&trainer->prop.input_meta, i);
gst_tensor_meta_info_convert (&in_meta[i], _info);
```

Reply:
Okay, I will use ```gst_tensors_info_get_nth_info```, thank you.